sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -1067,90 +1083,115 @@ public Object call(SQLDatabase db) throws Exception{
+                                        Object>) attachments.get(att)).get("data");
+                                InputStream is = Base64InputStreamFactory.get(new
+                                        ByteArrayInputStream(data.getBytes()));
+                                String type = (String) ((Map<String,
+                                        Object>) attachments.get(att)).get("content_type");
+                                // inline attachments are automatically decompressed,
+                                // so we don't have to worry about that
+                                UnsavedStreamAttachment usa = new UnsavedStreamAttachment(is,
+                                        att, type);
+                                try {
+                                    PreparedAttachment pa = prepareAttachment(usa);
+                                    attachmentManager.addAttachment(db, pa, rev);
+                                } catch (Exception e) {
+                                    logger.log(Level.SEVERE, "There was a problem adding the " +
+                                                    "attachment "
+                                                    + usa + "to the datastore for document " + rev,
+                                            e);
+                                    throw e;
                                 }
                             }
-                        } else {
+                        }
+                    } else {
 
-                            try {
-                                if (preparedAttachments != null) {
-                                    for (String[] key : preparedAttachments.keySet()) {
-                                        String id = key[0];
-                                        String rev = key[1];
+                        try {
+                            if (preparedAttachments != null) {
+                                for (String[] key : preparedAttachments.keySet()) {
+                                    String id = key[0];
+                                    String rev = key[1];
+                                    try {
                                         BasicDocumentRevision doc = getDocumentInQueue(db, id, rev);
-                                        if(doc != null) {
-                                            for (PreparedAttachment att : preparedAttachments.get(key)) {
+                                        if (doc != null) {
+                                            for (PreparedAttachment att : preparedAttachments.get
+                                                    (key)) {
                                                 attachmentManager.addAttachment(db, att, doc);
                                             }
                                         }
+                                    } catch (DocumentNotFoundException e){
+                                        continue;
```

Comment:
From my understanding it was expected that `getDocumentInQueue` could return null for the document at the specified revision and it was safe to continue. Since now  `getDocumentInQueue` throws `DocumentNotFoundException` instead of returning null it's fine just to continue to the loop. I'll add in a comment to explain this.